tokio/src/fs/file.rs

```diff
@@ -19,6 +20,18 @@ use std::task::Context;
 use std::task::Poll;
 use std::task::Poll::*;
 
+cfg_if! {
+    if #[cfg(test)] {
+        use super::mocks::MockFile as StdFile;
+        use super::mocks::JoinHandle;
+        use super::mocks::spawn_blocking;
+    } else {
+        use std::fs::File as StdFile;
+        use crate::blocking::spawn_blocking;
+        use crate::blocking::JoinHandle;
+    }
+}
```

Comment:
I'd rather not add a new dependency if this is all its used for. You can add new macros to `src/macros/cfg.rs` if you want something like this.

Reply:
It isn't actually a new dependency, because it's already a transitive dependency.  The getrandom, instant, log, loom, mockall, nix, parking_lot, tempfile, and tracing crates already depend on cfg-if.  Do you want me to remove it anyway?

Reply:
Yes, we already have a solution for conditional compilation blocks in `src/macros/cfg.rs`, so I would prefer to continue to use that rather than introduce a new convention for how to do it.

Besides, most of the crates you mention are dev-dependencies, so they don't count. There are feature combinations involving `fs` that do not currently depend on cfg-if, even transitively.

Reply:
removed.